src/app/spotify-api.service.ts

```diff
@@ -1,29 +1,304 @@
-import { Injectable, NgModule } from '@angular/core';
+import { Injectable, NgModule, OnInit } from '@angular/core';
 import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { DomSanitizer } from '@angular/platform-browser';
+import { Subject, BehaviorSubject, Observable, Subscription } from 'rxjs/Rx';
+import { LocalStorageService } from 'angular-2-local-storage';
+
+import { Artist } from './artist';
+import { Track } from './track';
+import qs from 'qs';
+import { Status } from './status';
 
 @Injectable()
 export class SpotifyApiService {
 
-  private apiUrl = 'https://api.spotify.com';
+  public artistSearchResults: Subject<Artist[]> = new Subject<Artist[]>();
+  public trackSearchResults: BehaviorSubject<Track[]> = new BehaviorSubject([new Track()]);
+  public currentArtist: BehaviorSubject<Artist> = new BehaviorSubject(new Artist());
+  public recentArtists: BehaviorSubject<Artist[]> = new BehaviorSubject([new Artist()]);
+  public user: BehaviorSubject<{}> = new BehaviorSubject({});
+  public playlist: BehaviorSubject<{}> = new BehaviorSubject({});
+  // public busy: Observable<any>;
+
+  private _artistString: string;
+  public status: Status;
+
+  private recentSize = 5; // number of recent artists to show
+  public playlistName = 'Duetify';
+  private playlistDescription = 'Duetify managed playlist of the last search artist';
+
+  private apiUrl = 'https://api.spotify.com/v1/';
   private access_token: string;
 
-  constructor(private http: HttpClient) { }
+  constructor(private http: HttpClient,
+    private sanitizer: DomSanitizer,
+    private localStorage: LocalStorageService
+  ) {
+    // this.getUser();
+    // this.getPlaylist();
+    this.user.subscribe(data => data['id'] && this.getPlaylist());
+
+    this.trackSearchResults.subscribe(tracks => (tracks && tracks.length > 0) ? this.reloadPlaylist() : undefined);
+    // Observable
```

Comment:
Try to deleted commented code before commiting or pushing (in the cleaning/refactor phase).